test/OpenTelemetry.Tests/Implementation/Trace/Propagation/CompositePropagatorTest.cs

```diff
@@ -0,0 +1,146 @@
+﻿// <copyright file="CompositePropagatorTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry.Context.Propagation;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Implementation.Trace.Propagation
+{
+    public class CompositePropagatorTest
+    {
+        private const string TraceParent = "traceparent";
+        private static readonly string[] Empty = new string[0];
+        private static readonly Func<IDictionary<string, string>, string, IEnumerable<string>> Getter = (headers, name) =>
+        {
+            if (headers.TryGetValue(name, out var value))
+            {
+                return new[] { value };
+            }
+
+            return Empty;
+        };
+
+        private static readonly Action<IDictionary<string, string>, string, string> Setter = (carrier, name, value) =>
+        {
+            carrier[name] = value;
+        };
+
+        private ActivityTraceId traceId = ActivityTraceId.CreateRandom();
+        private ActivitySpanId spanId = ActivitySpanId.CreateRandom();
+
+        [Fact]
+        public void CompositePropagator_EmptyTextFormatList()
+        {
+            Dictionary<string, string> expected = new Dictionary<string, string>();
+            Dictionary<string, string> carrier = new Dictionary<string, string>();
+            var compositePropagator = new CompositePropagator();
+            var activityContext = compositePropagator.Extract(default, carrier, Getter);
+            Assert.Equal(default, activityContext);
+
+            activityContext = new ActivityContext(this.traceId, this.spanId, ActivityTraceFlags.Recorded, traceState: null);
+            compositePropagator.Inject(activityContext, carrier, Setter);
+            Assert.Equal(expected, carrier);
+
+            bool isInjected = compositePropagator.IsInjected(carrier, Getter);
+            Assert.False(isInjected);
+        }
+
+        [Fact]
+        public void CompositePropagator_WithTraceContext()
```

Comment:
Composite propagator doesn't care what its composed of - to test it, we can write test propagators, and no use b3 or tracecontext.

Reply:
I created two new tests:
1. using a custompropagator + TraceContextFormat to check the injection and the extraction, the case where the CustomPropagator doesn't generate a valid ActivityContext and, the second one does
2. using CustomPropagator1 + CustomPropagator2 to check what would happen if both propagators sets the same variable.

Reply:
just updated!